### PR TITLE
S-89135: initial commit

### PR DESCRIPTION
**Pull request by rohithamroser**

### CODE REVIEW

File: github_plugin/transformer_functions.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
+import base64
+
+from external_plugins.github_integrate import default as DEFAULT
```
Comment (DhilipKumar0807) [inline github_plugin/transformer_functions.py]: change the github_integrate  to github_plugin

File: github_plugin/mapping.py [hunk 2]
```diff
@@ -9,14 +9,17 @@
     FieldDisplayIcon
 )
 
-from external_plugins.zendesk_plugin import transformer_functions
-import external_plugins.zendesk_plugin.default as DEFAULT
+from external_plugins.github_integrate import transformer_functions
```
Comment (DhilipKumar0807) [inline github_plugin/mapping.py]: same here change the github_integrate  to github_plugin

File: github_plugin/mapping.py [hunk 3]
```diff
@@ -9,14 +9,17 @@
...
+import external_plugins.github_integrate.default as DEFAULT
```
Comment (DhilipKumar0807) [inline github_plugin/mapping.py]: change the github_integrate  to github_plugin

File: github_plugin/sync.py [hunk 4]
```diff
@@ -14,52 +13,53 @@
 from datetime import datetime
 from dateutil import parser
 import time
-from external_plugins.zendesk_plugin import transformer_functions
+import json
+from external_plugins.github_integrate import transformer_functions
```
Comment (DhilipKumar0807) [inline github_plugin/sync.py]: change the github_integrate  to github_plugin

File: github_plugin/transformer_functions.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    return fields
+def connect(instance_details):
+   return ASyncRestApi(instance_details['url'],headers={
+                            "authorization": "Bearer {}".format(instance_details["token"]),
+                            "Accept": "application/json",
+                            "Content-Type": "application/json"
+                            })
+
+
+#def encode_to_base64_string(email, password):
```
Comment (DhilipKumar0807) [inline github_plugin/transformer_functions.py]: remove commented function if it is not used

File: github_plugin/transformer_functions.py [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+            DEFAULT.INITIAL_PATH, 
+            
+            )
+
+    response = instance.get(path)
+
+    if instance_details["Username"] == response["login"]:
+        return "Connection to Github server is successfull."
+    else:
+        return response
```
Comment (DhilipKumar0807) [inline github_plugin/transformer_functions.py]: log the response and send the proper failure message.

File: github_plugin/transformer_functions.py (hunk 1 above)
Comment (rohithamroser) [reply github_plugin/transformer_functions.py]: done

File: github_plugin/mapping.py (hunk 2 above)
Comment (rohithamroser) [reply github_plugin/mapping.py]: done

File: github_plugin/mapping.py (hunk 3 above)
Comment (rohithamroser) [reply github_plugin/mapping.py]: done

File: github_plugin/sync.py (hunk 4 above)
Comment (rohithamroser) [reply github_plugin/sync.py]: done

**Comment by ghost** [comment]
Merge

**Pull request merged**